### PR TITLE
Update pinot to 1.2.0

### PR DESCRIPTION
**Pull request by elonazoulay**
## Description

Update to pinot 1.2.0

Attempted to push to #23143 but could not find the branch. I manually copied @wendigo's first commit.

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java [hunk 1]
```diff
@@ -442,8 +442,8 @@ public TimeBoundary(
         {
             if (timeColumn != null && timeValue != null) {
                 // See org.apache.pinot.broker.requesthandler.BaseBrokerRequestHandler::attachTimeBoundary
-                offlineTimePredicate = Optional.of(format("%s <= %s", timeColumn, timeValue));
-                onlineTimePredicate = Optional.of(format("%s > %s", timeColumn, timeValue));
+                offlineTimePredicate = Optional.of(format("%s <= '%s'", timeColumn, timeValue));
```
Comment (elonazoulay) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotClient.java:445]: Note: this change also works with pinot 1.1.0. It was necessary as the type of the time boundary value changed from long to timestamp in pinot 1.2.0 (i.e. a timestamp literal), so single quotes are needed.

This works for both timestamp and long types now.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotModule.java [hunk 2]
```diff
@@ -120,6 +123,19 @@ public DataSchema deserialize(JsonParser p, DeserializationContext ctxt)
         }
     }
 
+    public static class BrokerResponseNativeDeserializer
```
Comment (elonazoulay) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/PinotModule.java:126]: Note: this was added due changes to the json serialization in this Pinot [pr](https://github.com/apache/pinot/pull/13179).

**Review by wendigo — APPROVED** [review]
We can add testing against Pinot 1.2 server as a follow-up

**Comment by wendigo** [comment]
I've squashed two commits into single one before merging

**Pull request merged**